### ISSUE TITLE
Build error Caddy v2.8.0 with Netcup

### ISSUE THREAD

**Issue by serfriz** [caddy-dns/netcup#5]
I am getting an error building Caddy v2.8.0 with this module (xcaddy) for the linux/amd64 platform using the following Dockerfile: https://github.com/serfriz/caddy-custom-builds/blob/main/caddy-netcup/Dockerfile

```
#34 [linux/ppc64le builder 2/2] RUN xcaddy build     --with github.com/caddy-dns/netcup
#34 203.1 go: downloading go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.49.0
#34 203.4 go: downloading github.com/zeebo/pcg v1.0.1
#34 203.5 go: downloading github.com/OneOfOne/xxhash v1.2.2
#34 203.7 go: downloading github.com/spaolacci/murmur3 v1.1.0
#34 203.9 go: downloading github.com/google/s2a-go v0.1.7
#34 206.0 go: downloading github.com/googleapis/enterprise-certificate-proxy v0.3.2
#34 206.6 go: downloading github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da
#34 220.1 2024/05/30 18:00:33 [INFO] exec (timeout=0s): /usr/local/go/bin/go build -o /usr/bin/caddy -ldflags -w -s -trimpath -tags nobadger 
#34 ...

#33 [linux/amd64 builder 2/2] RUN xcaddy build     --with github.com/caddy-dns/netcup
#33 233.8 2024/05/30 18:00:46 [INFO] Skipping cleanup as requested; leaving folder intact: /tmp/buildenv_2024-05-30-1756.4188945626
#33 233.8 2024/05/30 18:00:46 [FATAL] exit status 1
#33 ERROR: process "/bin/sh -c xcaddy build     --with github.com/caddy-dns/netcup" did not complete successfully: exit code: 1

#36 [linux/arm/v6 builder 2/2] RUN xcaddy build     --with github.com/caddy-dns/netcup
#36 CANCELED

#35 [linux/s390x builder 2/2] RUN xcaddy build     --with github.com/caddy-dns/netcup
#35 CANCELED

#31 [linux/arm64 builder 2/2] RUN xcaddy build     --with github.com/caddy-dns/netcup
#31 CANCELED

#32 [linux/arm/v7 builder 2/2] RUN xcaddy build     --with github.com/caddy-dns/netcup
#32 CANCELED

#34 [linux/ppc64le builder 2/2] RUN xcaddy build     --with github.com/caddy-dns/netcup
#34 CANCELED
------
 > [linux/amd64 builder 2/2] RUN xcaddy build     --with github.com/caddy-dns/netcup:
39.13 go: downloading go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.49.0
39.18 go: downloading github.com/google/s2a-go v0.1.7
39.50 go: downloading github.com/googleapis/enterprise-certificate-proxy v0.3.2
39.66 go: downloading github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da
40.95 2024/05/30 17:57:33 [INFO] exec (timeout=0s): /usr/local/go/bin/go build -o /usr/bin/caddy -ldflags -w -s -trimpath -tags nobadger 
114.4 # github.com/libdns/netcup
114.4 /go/pkg/mod/github.com/libdns/netcup@v0.1.0/util.go:27:14: cannot use record.Priority (variable of type int) as uint value in struct literal
114.4 /go/pkg/mod/github.com/libdns/netcup@v0.1.0/util.go:43:17: cannot use record.Priority (variable of type uint) as int value in struct literal
233.8 2024/05/30 18:00:46 [INFO] Skipping cleanup as requested; leaving folder intact: /tmp/buildenv_2024-05-30-1756.4188945626
233.8 2024/05/30 18:00:46 [FATAL] exit status 1
------
Dockerfile:4
--------------------
   3 |     
   4 | >>> RUN xcaddy build \
   5 | >>>     --with github.com/caddy-dns/netcup
   6 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c xcaddy build     --with github.com/caddy-dns/netcup" did not complete successfully: exit code: 1
Error: buildx failed with: ERROR: failed to solve: process "/bin/sh -c xcaddy build     --with github.com/caddy-dns/netcup" did not complete successfully: exit code: 1
```

**Comment by serfriz** [comment]
Issue solved thanks to the workaround from #4 implemented in the PR libdns/netcup#1. I replaced the libdns/netcup dependency with the patched one in the Dockerfile 3bc66fa1286beb5b8bf2641fca57a3ccac036b07 until the PR gets merged:

```
# syntax=docker/dockerfile:1
FROM caddy:2.8.1-builder AS builder

RUN xcaddy build \
    --with github.com/caddy-dns/netcup \
    --replace github.com/libdns/netcup=github.com/Monviech/libdns-netcup@libdns-patch

FROM caddy:2.8.1

COPY --from=builder /usr/bin/caddy /usr/bin/caddy
```

**Comment by serfriz** [comment]
> I would keep this issue open since its not resolved until there are new version tags for libdns/netcup and caddy-dns/netcup with updated dependencies.

I agree, reopening the issue.
